pkg/apis/core/v1/storagecluster.go

```diff
@@ -583,6 +584,23 @@ type PrometheusSpec struct {
 	// SecurityContext holds pod-level security attributes and common container settings.
 	// This defaults to the default PodSecurityContext. The value will pass through to Prometheus CR.
 	SecurityContext *v1.PodSecurityContext `json:"securityContext,omitempty"`
+	// Replicas specifies the number of replicas of each shard to deploy for a Prometheus deployment.
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Retention specifies time duration Prometheus shall retain data for.
+	Retention string `json:"retention,omitempty"`
+	// RetentionSize specifies maximum amount of disk space used by blocks.*
+	RetentionSize string `json:"retentionSize,omitempty"`
+	// StorageSpec defines the configured storage for a group Prometheus servers. If no storage option is
+	// specified, then by default an EmptyDir will be used. If multiple storage options are specified,
+	// priority will be given as follows: EmptyDir, Ephemeral, and lastly VolumeClaimTemplate.
+	Storage *monitoringv1.StorageSpec `json:"storage,omitempty"`
+	// Volumes allows configuration of additional volumes on the output StatefulSet definition. Volumes specified will
+	// be appended to other volumes that are generated as a result of StorageSpec objects.
+	Volumes []v1.Volume `json:"volumes,omitempty"`
+	// VolumeMounts allows configuration of additional VolumeMounts on the output StatefulSet definition.
+	// VolumeMounts specified will be appended to other VolumeMounts in the prometheus container,
+	// that are generated as a result of StorageSpec objects.
+	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
```

Comment:
do a `make codegen` locally to generate deep equal util functions for this crd.

Reply:
Added codegen change.